pkg/resources/storage_integration.go

```diff
@@ -52,12 +53,11 @@ var storageIntegrationSchema = map[string]*schema.Schema{
 		Optional:    true,
 		Description: "Explicitly prohibits external stages that use the integration from referencing one or more storage locations.",
 	},
-	// TODO (SNOW-1015282): Remove S3gov option before going into V1
 	"storage_provider": {
-		Type:         schema.TypeString,
-		Required:     true,
-		ForceNew:     true,
-		ValidateFunc: validation.StringInSlice([]string{"S3", "S3gov", "GCS", "AZURE", "S3GOV"}, false),
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		ValidateDiagFunc: StringInSlice(append(sdk.AsStringList(sdk.AllS3Protocols), "GCS", "AZURE"), true),
```

Comment:
We can also add a description here with a list of supported values.

Reply:
Done